stdlib/sys/fs.gr

```diff
@@ -0,0 +1,1068 @@
+  NoSuchProcess,
+  ConnectionTimedOut,
+  TextFileBusy,
+  CrossDeviceLink,
+  ExtensionCapabilitiesInsufficient,
+  UnknownFileError(Number),
+}
+
+let errnoToFileError = errno => {
+  match (errno) {
+    2 => PermissionDenied,
+    3 => AddressInUse,
+    4 => AddressNotAvailable,
+    5 => AddressFamilyNotSupported,
+    6 => ResourceUnavailableOrOperationWouldBlock,
+    7 => ConnectionAlreadyInProgress,
+    8 => BadFileDescriptor,
+    9 => BadMessage,
+    10 => DeviceOrResourceBusy,
+    11 => OperationCanceled,
+    12 => NoChildProcesses,
+    13 => ConnectionAborted,
+    14 => ConnectionRefused,
+    15 => ConnectionReset,
+    16 => ResourceDeadlockWouldOccur,
+    17 => DestinationAddressRequired,
+    18 => MathematicsArgumentOutOfDomainOfFunction,
+    20 => FileExists,
+    21 => BadAddress,
+    22 => FileTooLarge,
+    23 => HostIsUnreachable,
+    24 => IdentifierRemoved,
+    25 => IllegalByteSequence,
+    26 => OperationInProgress,
+    27 => InterruptedFunction,
+    28 => InvalidArgument,
+    29 => IOError,
+    30 => SocketIsConnected,
+    31 => IsADirectory,
+    32 => TooManyLevelsOfSymbolicLinks,
+    33 => FileDescriptorValueTooLarge,
+    34 => TooManyLinks,
+    35 => MessageTooLarge,
+    37 => FilenameTooLong,
+    38 => NetworkIsDown,
+    39 => ConnectionAbortedByNetwork,
+    40 => NetworkUnreachable,
+    41 => TooManyFilesOpenInSystem,
+    42 => NoBufferSpaceAvailable,
+    43 => NoSuchDevice,
+    44 => NoSuchFileOrDirectory,
+    45 => ExecutableFileFormatError,
+    46 => NoLocksAvailable,
+    48 => NotEnoughSpace,
+    49 => NoMessageOfTheDesiredType,
+    50 => ProtocolNotAvailable,
+    51 => NoSpaceLeftOnDevice,
+    52 => FunctionNotSupported,
+    53 => TheSocketIsNotConnected,
+    54 => NotADirectoryOrASymbolicLinkToADirectory,
+    55 => DirectoryNotEmpty,
+    56 => StateNotRecoverable,
+    57 => NotASocket,
+    58 => NotSupportedOrOperationNotSupportedOnSocket,
+    59 => InappropriateIOControlOperation,
+    60 => NoSuchDeviceOrAddress,
+    61 => ValueTooLargeToBeStoredInDataType,
+    62 => PreviousOwnerDied,
+    63 => OperationNotPermitted,
+    64 => BrokenPipe,
+    65 => ProtocolError,
+    66 => ProtocolNotSupported,
+    67 => ProtocolWrongTypeForSocket,
+    68 => ResultTooLarge,
+    69 => ReadOnlyFileSystem,
+    70 => InvalidSeek,
+    71 => NoSuchProcess,
+    73 => ConnectionTimedOut,
+    74 => TextFileBusy,
+    75 => CrossDeviceLink,
+    76 => ExtensionCapabilitiesInsufficient,
+    _ => UnknownFileError(errno),
+  }
+}
+
+/**
+ * Represents non-standard system file types.
+ */
+provide enum SpecialFileType {
+  Unknown,
+  BlockDevice,
+  CharacterDevice,
+  SocketDatagram,
+  SocketStream,
+}
+
+/**
+ * Represents different system file types.
+ */
+provide enum FileType {
+  File,
+  Directory,
+  SymbolicLink,
+  Special(SpecialFileType),
+}
+
+/**
+ * Represents metadata about a file.
+ */
+provide record Stats {
+  fileType: FileType,
+  size: Number,
```

Comment:
can we specify that size is in I assume bytes? idk if we want to document it using the proposed syntax in #1852 or just add the documentation for it to the record doc.